### PR TITLE
Add thailand; add code to correct some ZIPs

### PR DESCRIPTION
**Pull request by adamrp**
Gail noticed these problems in the most recent metadata, so adding code to correct the issues.  While working on this, found out about [Oracle's `regexp_like` function](http://docs.oracle.com/cd/B12037_01/server.101/b10759/conditions018.htm), which is neat and could come in handy in a lot of places!

### CODE REVIEW

**Comment by teravest** [comment]
Looks good Gail mentioned one other country, maybe the UK that was still missing. 
@ackermag  what countries besides Thailand were still broken?

File: database/procedures/ag_get_barcode_metadata.sql [hunk 1]
```diff
@@ -150,6 +150,7 @@ select akb.barcode as sample_name,
         end as state,
         case 
             when zip is null then 'unknown'
+            when regexp_like(zip, '^\d*$') then lpad(zip, 5, 0)
```
Comment (ElDeveloper) [inline database/procedures/ag_get_barcode_metadata.sql]: Nice clever solution @adamrp.

Comment (adamrp) [inline database/procedures/ag_get_barcode_metadata.sql]: Thanks!  Except that it's all wrong.  One moment please... Okay, pushed new commit :smile:

**Comment by adamrp** [comment]
UK is represented now, @teravest and @ackermag. The reason it was broken in the version of the metadata that I sent to Gail previously is that I forgot to actually put the latest code (which added a bunch of countries) on the live database server (I had it in the repo and deployed on Quarterbarrel, but never moved it over to live).  All should be well, now!

File: database/procedures/ag_get_barcode_metadata.sql (hunk 1 above)
Comment (ElDeveloper) [inline database/procedures/ag_get_barcode_metadata.sql]: Oh was your last change with the intent of supporting zip codes with more than 5 characters? What would be a good way to document this?

**Comment by rob-knight** [comment]
Thanks for resolving this, Adam!

On Apr 22, 2014, at 10:01 AM, adamrp <notifications@github.com<mailto:notifications@github.com>> wrote:

UK is represented now, @teravesthttps://github.com/teravest and @ackermaghttps://github.com/ackermag. The reason it was broken in the version of the metadata that I sent to Gail previously is that I forgot to actually put the latest code (which added a bunch of countries) on the live database server (I had it in the repo and deployed on Quarterbarrel, but never moved it over to live). All should be well, now!

File: database/procedures/ag_get_barcode_metadata.sql (hunk 1 above)
Comment (adamrp) [inline database/procedures/ag_get_barcode_metadata.sql]: Nah, it is intended to deal with 5-digit US zip codes (the majority of zip
codes we get) that have had leading zeros unduly stripped.  So the intent
is to identify these errors based on the criteria 1) that the zip code is
entirely numerical and 2) that the zip code is less than 5 digits.  Zips
containing non-numerical characters are probably from other countries
(which I don't know a standard way of attempting correction), or they could
be 9-digit US ZIPs (I could add logic for this case, but it's extremely
rare; what do you think?)

**Pull request merged**